File: BraintreeCore/src/main/java/com/braintreepayments/api/GraphQLQueryHelper.java
package com.braintreepayments.api;

import android.content.Context;
import android.content.res.Resources;

import java.io.IOException;
import java.io.InputStream;

class GraphQLQueryHelper {

    static String getQuery(Context context, int queryResource) throws Resources.NotFoundException, IOException {
        InputStream inputStream = null;
        try {
            inputStream = context.getResources().openRawResource(queryResource);
            return StreamHelper.getString(inputStream);
        } finally {
            if (inputStream != null) {
                inputStream.close();
            }
        }
    }
}

File: BraintreeCore/src/main/java/com/braintreepayments/api/GraphQLQueryHelper.kt
package com.braintreepayments.api

import android.content.Context
import android.content.res.Resources
import androidx.annotation.RestrictTo
import java.io.IOException
import java.io.InputStream

// NEXT_MAJOR_VERSION: remove class once its added to drop in
/**
 * @suppress
 */
@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
object GraphQLQueryHelper {
    @JvmStatic
    @Throws(Resources.NotFoundException::class, IOException::class)
    fun getQuery(context: Context, queryResource: Int): String {
        var inputStream: InputStream? = null
        return try {
            inputStream = context.resources.openRawResource(queryResource)
            StreamHelper.getString(inputStream)
        } finally {
            inputStream?.close()
        }
    }
}
